### PR TITLE
Temp fix for no child models getting restored in xAPI

### PR DESCRIPTION
**Pull request by RobertPeek**

### CODE REVIEW

File: js/adapt-assessmentArticleModel.js [hunk 1]
```diff
@@ -132,12 +132,82 @@ define([
...
+                if (questionBanksEnabled && !questionBanksLoaded) {
+
+                  this.getChildren().models = this._originalChildModels;
+                  if(assessmentConfig._banks &&
+                          assessmentConfig._banks._isEnabled &&
+                          assessmentConfig._banks._split.length > 1) {
+
+                      quizModels = this._setupBankedAssessment();
+                  } else if(assessmentConfig._randomisation &&
+                          assessmentConfig._randomisation._isEnabled) {
```
Comment (deltanetdan) [inline js/adapt-assessmentArticleModel.js]: Is this _banks randomisation or randomised blocks? If it's the later it is not dependent on `questionBanksEnabled && !questionBanksLoaded` and looks like a duplicate of the code below in the else statement.

File: js/adapt-assessmentArticleModel.js [hunk 2]
```diff
@@ -132,12 +132,82 @@ define([
...
+                            assessmentConfig._banks._isEnabled &&
+                            assessmentConfig._banks._split.length > 1) {
+
+                        quizModels = this._setupBankedAssessment();
+                    } else if(assessmentConfig._randomisation &&
+                            assessmentConfig._randomisation._isEnabled) {
+
+                        quizModels = this._setupRandomisedAssessment();
+                    }
+                    quizModels = this.getChildren().models;
```
Comment (deltanetdan) [inline js/adapt-assessmentArticleModel.js]: Isn't this just going to overwrite the two quizModels assignments above?

Comment (RobertPeek) [reply js/adapt-assessmentArticleModel.js]: I just hadn't removed the two quizModels assignments above so these are gone now.

File: js/adapt-assessmentArticleModel.js (hunk 1 above)
Comment (RobertPeek) [reply js/adapt-assessmentArticleModel.js]: It is randomised blocks so i've added a check for that in the if statement.

File: js/adapt-assessmentArticleModel.js [hunk 3]
```diff
@@ -132,12 +132,52 @@ define([
...
+            var questionBanksEnabled = this.get('_assessment')._banks._isEnabled;
+            var questionBanksLoaded = this._questionBanks;
+
+            var randomisedEnabled = this.get('_assessment')._randomisation._isEnabled;
+
             if (!quizModels) {
-                // leave the order as before, completed or not
-                quizModels = this.getChildren().models;
+
+              if ((questionBanksEnabled && !questionBanksLoaded) || randomisedEnabled) {
```
Comment (deltanetdan) [inline js/adapt-assessmentArticleModel.js:142]: on return from the menu quizModels is undefined but questionBanksLoaded is populated. So quizModels gets populated with all questions on line 153. 

Either quizModels should be pre-populated with this.getChildren().models (e.g. changing line 114 to
var quizModels = (this.getChildren().models.length > 0) ? this.getChildren().models : null; ) or there should be a condition for  "if ((questionBanksEnabled && questionBanksLoaded) || randomisedEnabled) {" that populates quizModels from the assessment model.

**Pull request merged**